### PR TITLE
Add node-addon-api async iterator example

### PR DESCRIPTION
**Pull request by KevinEady**
An example as requested by https://github.com/nodejs/node-addon-api/issues/1171

### CODE REVIEW

**Comment by KevinEady** [comment]
cc: @vmoroz @JckXia @deepakrkris

**Comment by KevinEady** [comment]
Hi team,

I realized that the first example I wrote is a bit contrived and probably not useful in real-world situations. I've created a new example that uses threadsafe functions which I think provides a better use-case in [d70e386](https://github.com/nodejs/node-addon-examples/pull/195/commits/d70e3865edf5e893cafeb67424a42aa3259c1944).

**Review by legendecas — APPROVED** [review]
LGTM 👍

**Review by mhdawson — APPROVED** [review]
LGTM

File: async-iterator/node-addon-api/example.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    });
+
+    iteratorObject["next"] =
+        next.Get("bind").As<Function>().Call(next, {iteratorObject});
+
+    return iteratorObject;
+  }
+
+ private:
+  int _from;
```
[the remark this answers is not in the document]
Comment (mhdawson) [reply async-iterator/node-addon-api/example.cc:75]: @KevinEady let us know what you think on this one.

[the remark this answers is not in the document]
Comment (KevinEady) [reply async-iterator/node-addon-api/example.cc:75]: Hi @legendecas / @mhdawson , 

It looks like the standard used in `napi.h` for non-public members is to prefix with underscore (46 occurrences) versus suffix (1 occurrence), so I think I will stick with the prefix.

[the remark this answers is not in the document]
Comment (mhdawson) [reply async-iterator/node-addon-api/example.cc:75]: +1

**Pull request merged**